instrumentation/reactor-netty/reactor-netty-0.9/javaagent/src/test/groovy/AbstractReactorNettyHttpClientTest.groovy

```diff
@@ -52,4 +58,51 @@ abstract class AbstractReactorNettyHttpClientTest extends HttpClientTest<HttpCli
   }
 
   abstract HttpClient createHttpClient()
+
+  def "should expose context to http client callbacks"() {
+    given:
+    def onRequestSpan = new AtomicReference<Span>()
+    def afterRequestSpan = new AtomicReference<Span>()
+    def onResponseSpan = new AtomicReference<Span>()
+    def afterResponseSpan = new AtomicReference<Span>()
+
+    def httpClient = createHttpClient()
+      .doOnRequest({ rq, con -> onRequestSpan.set(Span.current()) })
+      .doAfterRequest({ rq, con -> afterRequestSpan.set(Span.current()) })
+      .doOnResponse({ rs, con -> onResponseSpan.set(Span.current()) })
+      .doAfterResponse({ rs, con -> afterResponseSpan.set(Span.current()) })
+
+    when:
+    runUnderTrace("parent") {
+      httpClient.baseUrl(server.address.toString())
+        .get()
+        .uri("/success")
+        .response()
+        .block()
+    }
+
+    then:
+    assertTraces(1) {
+      trace(0, 3) {
+        def parentSpan = span(0)
+        def nettyClientSpan = span(1)
+
+        basicSpan(it, 0, "parent")
+        clientSpan(it, 1, parentSpan, "GET", server.address.resolve("/success"))
+        serverSpan(it, 2, nettyClientSpan)
+
+        assertSameSpan(parentSpan, onRequestSpan)
+        assertSameSpan(nettyClientSpan, afterRequestSpan)
+        assertSameSpan(nettyClientSpan, onResponseSpan)
+        assertSameSpan(nettyClientSpan, afterResponseSpan)
```

Comment:
I like the reactor-netty-1.0 behavior better where `afterResponseSpan` uses the `parentSpan` instead of the `nettyClientSpan`, as it seems to match our general practice of response handlers being outside of the client span, and running in the parent context

if that's not something easy to change in 0.9 that's ok, would be good to add comment somewhere about this difference

Reply:
Done.